exla/lib/exla/defn.ex

```diff
@@ -733,6 +733,44 @@ defmodule EXLA.Defn do
     )
   end
 
+  defp to_operator(:take_along_axis, [tensor, indices, axis], _ans, state) do
+    indices_shape = op_shape(indices)
+    indices_rank = tuple_size(indices_shape)
+
+    axes_range = 0..(indices_rank-1)
```

Comment:
Can the rank be zero? If so, this needs an explicit step:

```suggestion
    axes_range = 0..(indices_rank-1)//1
```

Reply:
Good question, let me check

Reply:
Nope, there's a validation for this in Nx.Shape.
Also, it would be just the same as `Nx.concatenate([tensor ...])`, so we don't lose anything by not allowing this